### PR TITLE
Small moves: part three

### PR DESCRIPTION
**Pull request by keirog**
No refactoring yet! Just:
- lots of linting
- moved from travis to circleci
- minor clean-up of the config file

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[**{.js,.ts,.html,.scss,.css,.json,.yml,.yaml}]
+charset = utf-8
+end_of_line = lf
+indent_style = tab
```
Comment (AbdullahAli) [inline .editorconfig]: 🙉

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+module.exports = {
+	"env": {
+		"browser": false,
+		"es6": true,
+		"node": true,
+		"mocha": true
+	},
+	"extends": "airbnb-base",
```
Comment (AbdullahAli) [inline .eslintrc.js]: 👍🏼

File: models/route.js [hunk 3]
```diff
@@ -1,58 +1,50 @@
...
 
-					var name = tokens[1];
-					var host = tokens[2];
+					const name = tokens[1];
+					const host = tokens[2];
 
 					if (service.name === name) {
-						debug("Client requested a canary for service %s", service.name);
-						version.nodes = ['http://' + host];
+						debug('Client requested a canary for service %s', service.name);
```
Comment (AbdullahAli) [inline models/route.js]: Can we change the `%s` with ```` please

File: models/service.js [hunk 4]
```diff
@@ -1,19 +1,16 @@
...
-//  Figure out which service is the default
+// TODO - move to service model
+// Figure out which service is the default
 
 function getDefaultServiceVersion(versions) {
-	var defaultService = _.find(versions, function(service) {
-		var isPrimary = service.isPrimary ? true : false;
-		return isPrimary;
+	const defaultService = _.find(versions, (service) => {
+		return !!service.isPrimary;
```
Comment (AbdullahAli) [inline models/service.js]: Can we one line it please 👍 

` const defaultService = _.find(versions, service =>  !!service.isPrimary);`

File: models/service/collection.js [hunk 5]
```diff
@@ -1,27 +1,19 @@
...
+	this.filterByPath = function filterByPath(path) {
+		return _.first(this.profiles.filter((profile) => {
 
-		// TODO - path is deprecated, use paths only in the future
-		var pathsRegEx = profile.paths ? profile.paths : [profile.path];
-		return _.find(pathsRegEx, function(pathRegEx) {
-			return (new RegExp(pathRegEx)).test(path);
-		});
+			// TODO - path is deprecated, use paths only in the future
+			const pathsRegEx = profile.paths ? profile.paths : [profile.path];
```
Comment (AbdullahAli) [inline models/service/collection.js]: Can we change this please:

`const pathsRegEx = profile.paths || [profile.path];`

**Comment by AbdullahAli** [comment]
>No refactoring yet! 

Ah sorry, I just read this bit - please feel free to ignore my review 👍

File: models/route.js (hunk 3 above)
Comment (keirog) [reply models/route.js]: I would but there’s another PR that removes `debug` 😄

File: models/service/collection.js (hunk 5 above)
Comment (keirog) [reply models/service/collection.js]: Ah yep, a lot of this is auto-fixed 😅

**Pull request merged**